src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -27,6 +30,18 @@ public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider
             _services = services;
             _next = next;
         }
+        
+        public IServiceScopeFactory ScopeFactory
+        {
+            get
+            {
+                if (_scopeFactory == null)
+                {
+                    _scopeFactory = _services.GetRequiredService<IServiceScopeFactory>();
```

Comment:
We should probably verify this is safe to do with 3rd party containers. It's supposed to be but I'm not sure if it's part of our spec tests.

/cc @pranavkm